src/base/dev/ne2k/ne2000.c

```diff
@@ -670,7 +670,7 @@ static inline void ne2000_mem_writew(NE2000State *s, uint32_t addr,
 
     addr &= ~1; /* XXX: check exact behaviour if not even */
     if (addr < 32 ||
-        (addr >= NE2000_PMEM_START && addr < NE2000_MEM_SIZE)) {
+        (addr >= NE2000_PMEM_START && addr < (NE2000_MEM_SIZE - 2))) {
```

Comment:
I believe these all should be `<=`.

Reply:
Interesting, I just checked the latest qemu code and they seem to have one of each
https://github.com/qemu/qemu/blob/23901b2b721c0576007ab7580da8aa855d6042a9/hw/net/ne2000.c#L473-L494

Reply:
They do it correctly.